langchain/src/output_parsers/expression.ts

```diff
@@ -22,20 +21,31 @@ import { ASTParser } from "./expression_type_handlers/base.js";
  */
 
 export class ExpressionParser extends BaseOutputParser<ParsedType> {
+  parser: ParseFunction;
+
+  /**
+   * We should separate loading the parser into its own function
+   * because loading the grammar takes some time. If there are
+   * multiple concurrent parse calls, it's faster to just wait
+   * for building the parser once and then use it for all
+   * subsequent calls. See expression.test.ts for an example.
+   */
+  async loadParser() {
```

Comment:
We've called methods like this `ensureParser()` and wrapped it in an `if (!this.parser)` statement.

Reply:
Noted. I'll rename it.